app/src/main/java/com/jjswigut/matters/ui/EditMatterFragment.kt

```diff
@@ -34,26 +30,32 @@ class EditMatterFragment : BaseFragment() {
         super.onViewCreated(view, savedInstanceState)
 
         arguments?.let {
-            matter = EditMatterFragmentArgs.fromBundle(it).matter
+            if (it.containsKey(ARG_MATTER)) {
+                matter = it.getSerializable(ARG_MATTER) as Matter
+            }
             binding.titleInput.setText(matter?.matterTitle)
             binding.contentInput.setText(matter?.matterContent)
         }
 
         binding.saveFab.setOnClickListener {
-            input()
-
             launch {
 
                 context?.let {
-                    val mMatter = Matter(input().matterTitle, input().matterContent)
-
+                    // if matter is null, we had no arguments to start out fragment
+                    val title = binding.titleInput.text.toString().trim()
```

Comment:
so the flow of the logic is 

1. grab the input
2. check if our matter object is null - if it is, that means we are creating a new one - insert the new matter
3. if it is not, that means we are editing an existing one, so update the title / content of that existing one with the input from step 1 - update existing matter